experiment/secret-sync-controller/main.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"k8s.io/test-infra/experiment/secret-sync-controller/client"
+	"log"
+	"os"
+	"reflect"
+)
+
+type options struct {
+	configPath string
+}
+
+func (o *options) Validate() error {
+	if o.configPath == "" {
+		return errors.New("required flag --config-path was unset")
+	}
+
+	return nil
+}
+
+func gatherOptions() options {
+	o := options{}
+	flag.StringVar(&o.configPath, "config-path", "", "Path to config.yaml.")
+	flag.Parse()
+	return o
+}
+
+func main() {
+	o := gatherOptions()
+
+	// TODO: modularize clients
+	k8sClientset, err := client.NewK8sClientset()
+	if err != nil {
+		log.Fatal(err)
+	}
+	secretManagerClient, err := client.NewSecretManagerClient(context.Background())
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var clientInterface ClientInterface
+
+	client := Client{
+		K8sClientset:        k8sClientset,
+		SecretManagerClient: secretManagerClient,
+	}
+
+	clientInterface = client
+
+	secretSyncConfig, err := LoadConfig(o.configPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for i, spec := range secretSyncConfig.Specs {
+		fmt.Printf("Secret pair [%d]:\n{\n", i)
+		err := Sync(clientInterface, spec)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
```

Comment:
I would suggest pulling this out of main and into the controller struct I mentioned elsewhere.  Consider adding a flag to gate one-shot mode vs. running continuously

Reply:
This is now implemented in the Controller struct.
```
type Controller struct {
	Client       client.Interface
	Config       *SecretSyncConfig
	RunOnce      bool
	ResyncPeriod time.Duration
}
```
`RunOnce` stands for one-shot mode.